zamba/models/model_manager.py

```diff
@@ -104,60 +120,88 @@ def instantiate_model(
+                scheduler_config=scheduler_config,
+                hparams=hparams,
+                species=species,
+                model_class=model_class,
+                checkpoint=checkpoint,
             )
 
-        # update in case we want to finetune with different scheduler
-        if scheduler_config != "default":
-            hparams.update(scheduler_config.dict())
+    # without a subset, you will always get a new head
+    # the config validation prohibits setting use_default_model_labels to True without a subset
+    else:
+        logger.info(
+            "Provided species do not fully overlap with Zamba species. Replacing model head and finetuning."
+        )
+        return replace_head(
+            scheduler_config=scheduler_config,
+            hparams=hparams,
+            species=species,
+            model_class=model_class,
+            checkpoint=checkpoint,
+        )
 
-        hparams.update({"species": species})
-        model = model_class(finetune_from=checkpoint, **hparams)
 
+def replace_head(scheduler_config, hparams, species, model_class, checkpoint):
+    # update in case we want to finetune with different scheduler
+    if scheduler_config != "default":
+        hparams.update(scheduler_config.dict())
+
+    hparams.update({"species": species})
+    model = model_class(finetune_from=checkpoint, **hparams)
+    log_schedulers(model)
+    return model
+
+
+def resume_training(
+    scheduler_config,
+    hparams,
+    species,
+    model_class,
+    checkpoint,
+    labels,
+):
     # resume training; add additional species columns to labels file if needed
-    elif is_subset:
-        logger.info(
-            "Provided species fully overlap with Zamba species. Resuming training from latest checkpoint."
-        )
-        # update in case we want to resume with different scheduler
-        if scheduler_config != "default":
-            hparams.update(scheduler_config.dict())
+    logger.info(
+        "Provided species fully overlap with Zamba species. Resuming training from latest checkpoint."
+    )
+    # update in case we want to resume with different scheduler
+    if scheduler_config != "default":
+        hparams.update(scheduler_config.dict())
+
+    model = model_class.load_from_checkpoint(checkpoint_path=checkpoint, **hparams)
 
-        model = model_class.load_from_checkpoint(checkpoint_path=checkpoint, **hparams)
+    # add in remaining columns for species that are not present
+    for c in set(hparams["species"]).difference(set(species)):
+        # labels are still OHE at this point
+        labels[f"species_{c}"] = 0
 
-        # add in remaining columns for species that are not present
-        for c in set(hparams["species"]).difference(set(species)):
-            # labels are still OHE at this point
-            labels[f"species_{c}"] = 0
+    # sort columns so columns on dataloader are the same as columns on model
+    labels.sort_index(axis=1, inplace=True)
```

Comment:
Is there a simple way to get the order from the base model and just use that rather than assuming alphabetical?